scipy/stats/_stats_py.py

```diff
@@ -259,12 +259,6 @@ def gmean(a, axis=0, dtype=None, weights=None):
     The geometric average is computed over a single dimension of the input
     array, axis=0 by default, or all values in the array if axis=None.
     float64 intermediate and return values are used for integer inputs.
-    Beginning in SciPy 1.9, ``np.matrix`` inputs are converted to
-    ``np.ndarray``s before the calculation is performed. In this case, the
-    output will be a scalar or ``np.ndarray`` of appropriate shape rather than
-    a 2D ``np.matrix``. Similarly, while masked elements of masked arrays
-    are still ignored, the output will be a scalar or ``np.ndarray`` rather
-    than a masked array with ``mask=False``.
```

Comment:
Remove from `gmean` in particular and apply to all functions that get the decorator.

Reply:
Looks good!